examples/checkout.c

```diff
@@ -0,0 +1,102 @@
+#include "common.h"
+#include <git2.h>
+#include <git2/checkout.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#ifndef _WIN32
+# include <pthread.h>
+# include <unistd.h>
+#endif
+
+
+typedef struct progress_data {
+    size_t completed_steps;
+    size_t total_steps;
+    const char *path;
+} progress_data;
+
+
+static void print_progress(const progress_data *pd)
+{
+    int checkout_percent = pd->total_steps
+                           ? (100 * pd->completed_steps / pd->total_steps)
+                           : 100;
+
+    printf( "chk %3d%% (%4" PRIuZ "/%4" PRIuZ ") %s\n",
+            checkout_percent,
+            pd->completed_steps, pd->total_steps,
+            pd->path );
+}
+
+
+static void checkout_progress(const char *path, size_t cur, size_t tot, void *payload)
+{
+    progress_data *pd = (progress_data*)payload;
+    pd->completed_steps = cur;
+    pd->total_steps = tot;
+    pd->path = path;
+    print_progress(pd);
+}
+
+
+int do_checkout_ref(git_repository *repo, git_reference *checkout_ref)
+{
+    progress_data pd = {0};
+    git_object *target_tree = NULL;
+    git_checkout_options checkout_opts = GIT_CHECKOUT_OPTIONS_INIT;
+
+    int error;
+
+    // Set up options
+    checkout_opts.checkout_strategy = GIT_CHECKOUT_SAFE | GIT_CHECKOUT_ALLOW_CONFLICTS;
+    checkout_opts.progress_cb = checkout_progress;
+    checkout_opts.progress_payload = &pd;
+
+    // Get the tree for checkout
+    error = git_reference_peel( &target_tree, checkout_ref, GIT_OBJ_TREE );
+    if ( error ) {
+        goto error;
+    }
```

Comment:
Please avoid spaces in braces. So it should rather be `if (error)`. Furthermore, there is no need for curly braces around single-line statements.